addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,339 @@
+/**
+ * @author RT_Borg
+ *
+ * Uses topic id as a proxy for age, which doesn't require loading other pages
+ * to read actual dates.
+ */
+
+export default async function ({ addon, global, console }) {
+  // Default is a little over a month's worth of topics. Reasonably stable through 2022.
+  // There's a lot of tolerance. Half or twice as many topics filter pretty much the same.
+  const defaultTopicInterval = 7500;
+  const defaultForums = [
+    "New Scratchers",
+    "Help with Scripts",
+    "Questions about Scratch",
+    "Bugs and Glitches",
+    "Project Ideas",
+  ];
+  // Forums beyond these seem very poor options,
+  // as many posts live long or are appropriately revived
+  const customForumSettingToForum = {
+    applyToNewScratchers: "New Scratchers",
+    applyToHelpWithScripts: "Help with Scripts",
+    applyToQuestionsAboutScratch: "Questions about Scratch",
+    applyToBugsAndGlitches: "Bugs and Glitches",
+    applyToProjectIdeas: "Project Ideas",
+    applyToRequests: "Requests",
+    applyToSuggestions: "Suggestions",
+    applyToOpenSourceProjects: "Open Source Projects",
+  };
+  const searchResultsPageName = "Search Results"; // recent posts and unanswered posts
+  const isMobileSite = await determineIsMobileSite();
+  // tableBodyNode will be null on mobile and non-table search page:
```

Comment:
We don't need to use a function for something we only call once!
```suggestion
  const searchResultsPageName = "Search Results"; // recent posts and unanswered posts
  const isMobileSite = /^\/discuss\/m\//.test(location.pathname)
  // tableBodyNode will be null on mobile and non-table search page:
```

Reply:
Using a function can make the code more readable and organized. In this case the variable name is clear enough though.